### PR TITLE
Squashed commit of the following:

### PR DESCRIPTION
**Pull request by RyuMaster**
Changes needed for cross-compile to windows. Compiling done with MSYS2/MINGw64

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -1,3 +1,7 @@
-ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS} -Im4
+ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS} -I m4
 
+if WINDOWS
+SUBDIRS = xayagame
+else
 SUBDIRS = xayagame mover
+endif
```
Comment (domob1812) [inline Makefile.am]: Why do you disable mover for Windows?  Does it not cross-compile yet (and if so, do you think it would be particularly hard to fix)?  Note that even if you have your own implementation in C#, there is nothing wrong with compiling the C++ version `moverd` as well on Windows.  And if possible, I would like to reduce the number of platform-conditional things to the absolute minimum.

File: configure.ac [hunk 2]
```diff
@@ -7,14 +7,52 @@ AC_CONFIG_HEADERS([config.h])
...
+
 AC_LANG([C++])
 LT_INIT
 
+# Windows mu have no undefined symbol in dll, we need -no-undefined 
+# flag to work, so lets trigger clean dll build
+
+AM_COND_IF( [WINDOWS], [
+LT_INIT([win32-dll])
+])
```
Comment (domob1812) [inline configure.ac]: In the comment, you mention the `-no-undefined` flag - how is that related to the actual command later on?  Does `win32-dll` set this?  Also, as a small nit, please remove the blank line between the comment and the code below.

File: configure.ac [hunk 3]
```diff
@@ -7,14 +7,52 @@ AC_CONFIG_HEADERS([config.h])
...
+
+
+
```
Comment (domob1812) [inline configure.ac]: Please don't add these blank lines as well as more below.  Ideally, just have a single blank line between different blocks of statements.

File: configure.ac [hunk 4]
```diff
@@ -26,12 +64,29 @@ PKG_CHECK_MODULES([SQLITE3], [sqlite3],
...
 PKG_CHECK_MODULES([ZMQ], [libzmq])
 
+# When cross-compile to windows, we don't need mover folder
+# also we need to pass extra CPP flag for jsonrpccpp library
+# CONDITIONAL START
+  
+
+
+AM_COND_IF( [WINDOWS], [
+CXXFLAGS+=" -DGLOG_NO_ABBREVIATED_SEVERITIES"
```
Comment (domob1812) [inline configure.ac]: This is something that we should add generally, not conditioned on Windows.  While this solves a problem particular to Windows, the code does not need the abbreviated severities anyway, so we should disable them in general.  So just add this to `CXXFLAGS` unconditionally.

File: xayagame/Makefile.am [hunk 5]
```diff
@@ -3,13 +3,27 @@ xayagamedir = $(includedir)/xayagame
...
   -lstdc++fs
+  
+
+# When building for windows, jsonrpccpp library can only be build without hredis;
+# So we need to make sure, the -lhredis flag is not passed
+# Also, lets pass -no-undefined to make sure we have zero undefined symbols
+# Else mingw will not be able to build shared libraries
+if WINDOWS
+  libxayagame_la_LDFLAGS = $(JSONCPP_LDFLAGS) $(JSONRPCCLIENT_LDFLAGS) $(JSONRPCSERVER_LDFLAGS) \
+  $(GLOG_LDFLAGS) $(SQLITE3_LDFLAGS) $(ZMQ_LDFLAGS) -no-undefined
```
Comment (domob1812) [inline xayagame/Makefile.am]: Is adding `-no-undefined` here related in any way to the `win32-dll` thing from above?  Do you need both?  Also there seems to be nothing related to `hredis` in the actual code, you just mention it in the comment.  Is this an oversight?

File: xayagame/Makefile.am [hunk 6]
```diff
@@ -42,34 +56,39 @@ xayagame_HEADERS = \
   rpc-stubs/xayarpcclient.h \
   rpc-stubs/xayarpcserverstub.h
 
-check_PROGRAMS = tests
-TESTS = tests
+if WINDOWS
```
Comment (domob1812) [inline xayagame/Makefile.am]: Why do you disable the tests on Windows?  Are they hard to cross-compile?

File: xayagame/game.hpp [hunk 7]
```diff
@@ -13,7 +13,11 @@
 #include "uint256.hpp"
 #include "zmqsubscriber.hpp"
 
-#include "rpc-stubs/xayarpcclient.h"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated"
+ #include "rpc-stubs/xayarpcclient.h"
+#pragma GCC diagnostic pop
+
```
Comment (domob1812) [inline xayagame/game.hpp]: Why is this needed?  I'd rather not use `#pragma` if not absolutely necessary.

File: Makefile.am (hunk 1 above)
Comment (RyuMaster) [reply Makefile.am]: My target goal was to cross-compile core library only, not sure why anyone would need mover to run on windows. It had some compiling troubles, but yes, if you think its best to get rid of as much platform-conditional things as possible, I will get rid of those and cross-compile with mover.

File: configure.ac (hunk 3 above)
Comment (RyuMaster) [reply configure.ac]: Ok!

File: xayagame/Makefile.am (hunk 6 above)
Comment (RyuMaster) [reply xayagame/Makefile.am]: My target goal was to cross-compile core library only, not sure why anyone would need tests to run on windows. But yes, if you think its best to get rid of as much platform-conditional things as possible, I will get rid of those and cross-compile with test.

File: configure.ac (hunk 2 above)
Comment (RyuMaster) [reply configure.ac]: I am not sure, I remember I only got correct results with combination of win32-dll being set there and -no-undefined later on. I will try removing those now and see, if that is really necessary. If so, so I will supply error log with explanations.

File: configure.ac (hunk 3 above)
Comment (RyuMaster) [reply configure.ac]: Ok!

File: xayagame/Makefile.am (hunk 5 above)
Comment (RyuMaster) [reply xayagame/Makefile.am]: I am not sure, I remember I only got correct results with combination of win32-dll being set there and -no-undefined later on. I will try removing those now and see, if that is really necessary. If so, so I will supply error log with explanations. As for the 'hredis', I resolved it by directly modifying SON-RPC-CPP source, and so its "resolved" outsided this project scope. They are going to merge my fix soon.

File: configure.ac (hunk 4 above)
Comment (RyuMaster) [reply configure.ac]: Ok!

File: xayagame/game.hpp (hunk 7 above)
Comment (RyuMaster) [reply xayagame/game.hpp]: As we have: `AX_CHECK_COMPILE_FLAG([-Werror], [CXXFLAGS+=" -Werror"])`
this is not compiling, as xayarpcclient.h has depricated code for c++14, and in c++17 it can not compile at all anymore. I was getting this problem on all of my machines, so I decided to put this workaround, rather then conditionally removing  `AX_CHECK_COMPILE_FLAG([-Werror], [CXXFLAGS+=" -Werror"])`

File: Makefile.am (hunk 1 above)
Comment (domob1812) [reply Makefile.am]: Yes, please try to compile the entire library for Windows.  The C++ mover could either be used by people directly who want to play around with it, or it could be used as starting point for someone who wants to write a game in C++ on Windows.  Both seem useful - so unless there are particular problems with cross-compiling it (which I don't expect), it seems simpler to just compile everything than add conditions.

File: configure.ac (hunk 2 above)
Comment (domob1812) [reply configure.ac]: Ok.  If you need indeed both, then let me know.  I don't think you need to post detailed error logs - as long as you are confident that both are needed, I'll trust you on that. :)

File: xayagame/Makefile.am (hunk 5 above)
Comment (domob1812) [reply xayagame/Makefile.am]: Ok, thanks for fixing the hredis thing upstream!  Then just remove its mention from the comment, as it is misleading otherwise.

File: xayagame/Makefile.am (hunk 6 above)
Comment (domob1812) [reply xayagame/Makefile.am]: If possible, please compile also the tests.  Ideally, they should even *run* without issues - that's why we have them in the first place! :D

File: xayagame/game.hpp (hunk 7 above)
Comment (domob1812) [reply xayagame/game.hpp]: I don't get those errors, strangely ... but perhaps it depends on the compiler.  Perhaps just add `-Wno-deprecated` to the compile flags - I think I like that better than using a `#pragma`.

File: xayagame/Makefile.am (hunk 5 above)
Comment (RyuMaster) [reply xayagame/Makefile.am]: "Then just remove its mention from the comment, as it is misleading otherwise." - good point, will do

File: xayagame/Makefile.am (hunk 6 above)
Comment (RyuMaster) [reply xayagame/Makefile.am]: ok! =)
